test/table.js

```diff
@@ -1390,6 +1390,238 @@ describe('Bigtable/Table', function() {
+
+          // Checks that clearInterval was called.
+          setTimeoutSpy.called;
+          assert.ifError(err);
+          assert.strictEqual(response, true);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return false after 10 min if inconsistency repeats', done => {
+        table.bigtable.request = function(config, callback) {
+          if (config.method === 'generateConsistencyToken') {
+            return callback(null, {consistencyToken: 'sample-token12345'});
+          }
+          if (config.method === 'checkConsistency') {
+            return callback(null, {consistent: false});
+          }
+        };
+
+        table.waitForReplication(function(err, response) {
+          setTimeoutSpy.called;
+          assert.strictEqual(response, false);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return error if checkonsistency returns error', done => {
+        let error = new Error('consistency-check error');
+
+        responses = [
+          (config, callback) =>
+            callback(null, {consistencyToken: 'sample-token12345'}),
+          (config, callback) => callback(error),
+        ];
+
+        table.waitForReplication((err, res) => {
+          clearTimeoutSpy.called;
+          assert.strictEqual(checkConsistencySpy.callCount, 1);
+          assert.strictEqual(err, error);
+          assert.strictEqual(res, undefined);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('generateConsistencyToken', function() {
+    it('should provide proper request options', function(done) {
+      table.bigtable.request = function(config) {
+        assert.strictEqual(config.client, 'BigtableTableAdminClient');
+        assert.strictEqual(config.method, 'generateConsistencyToken');
+        assert.strictEqual(config.reqOpts.name, table.name);
+        done();
+      };
+      table.generateConsistencyToken(assert.ifError);
+    });
+
+    it('should return a consistencyToken', function(done) {
+      const cToken = 'sample-token-123456';
+      const response = {
+        consistencyToken: cToken,
+      };
+
+      table.bigtable.request = function(config, callback) {
+        callback(null, response);
+      };
+
+      table.generateConsistencyToken(function(err, token) {
+        assert.strictEqual(token, cToken);
+        done();
+      });
+    });
+  });
+
+  describe('checkConsistency', function() {
+    it('should provide the proper request options', function(done) {
+      const cToken = 'consistency-token-123';
+
+      table.bigtable.request = function(config) {
+        assert.strictEqual(config.client, 'BigtableTableAdminClient');
+        assert.strictEqual(config.method, 'checkConsistency');
+        assert.strictEqual(config.reqOpts.name, table.name);
+        assert.strictEqual(config.reqOpts.consistencyToken, cToken);
+        done();
+      };
+
+      table.checkConsistency(cToken);
```

Comment:
You can do `table.checkConsistency(cToken, assert.ifError);`, so that it catches an error.